credentials/alts/utils.go

```diff
@@ -83,28 +82,27 @@ var (
 
 // isRunningOnGCP checks whether the local system, without doing a network request is
 // running on GCP.
-func isRunningOnGCP() bool {
+func isRunningOnGCP() (bool, error) {
 	manufacturer, err := readManufacturer()
 	if os.IsNotExist(err) {
-		return false
+		return false, nil
 	}
 	if err != nil {
-		log.Fatalf("failure to read manufacturer information: %v", err)
+		return false, status.Errorf(codes.Internal, "failed to read manufacturer information: %v", err)
 	}
 	name := string(manufacturer)
 	switch runningOS {
 	case "linux":
 		name = strings.TrimSpace(name)
-		return name == "Google" || name == "Google Compute Engine"
+		return name == "Google" || name == "Google Compute Engine", nil
 	case "windows":
 		name = strings.Replace(name, " ", "", -1)
 		name = strings.Replace(name, "\n", "", -1)
 		name = strings.Replace(name, "\r", "", -1)
-		return name == "Google"
+		return name == "Google", nil
 	default:
-		log.Fatal(platformError(runningOS))
+		return false, nil
```

Comment:
Should we return `platformError(runningOS)` instead of `nil`?

Reply:
If we return an error, these errors will still be captured by tests. If MacOS users run the tests, I think the tests would still fail? Are there any ways that we can solve this issue(since our original goal is to let MacOS users to be able to run tests in OSS)?

Reply:
Ah I see, that's a good point. Then not returning an error is fine with me.